src/main/java/com/minecolonies/api/colony/requestsystem/requestable/StackList.java

```diff
@@ -431,4 +441,48 @@ public Set<TypeToken<?>> getSuperClasses()
     {
         return TYPE_TOKENS;
     }
+
+    public StackList(@NotNull final TagKey<Item> tag,
+        @NotNull final ServerLevel level,
+        final String description,
+        final int count,
+        final int minCount,
+        final int leftOver)
+    {
+        this(tagToStacks(tag, level.registryAccess(), count), description, count, minCount, leftOver);
+    }
+
+
+    /**
+     * Transforms a given tag into a list of itemstacks, with each
+     * itemstack having a count of perStackCount.
+     *
+     * @param tag the tag to transform.
+     * @param registryAccess the registry access.
+     * @param perStackCount the count of each itemstack.
+     * @return the resulting list of itemstacks.
+     */
+    private static List<ItemStack> tagToStacks(@NotNull final TagKey<Item> tag,
+        @NotNull final RegistryAccess registryAccess,
+        final int perStackCount)
+    {
+        final Registry<Item> itemReg = registryAccess.registryOrThrow(Registries.ITEM);
+        final Optional<HolderSet.Named<Item>> opt = itemReg.getTag(tag);
+
+        if (opt.isEmpty())
+        {
+            return List.of();
+        }
+
+        return opt.get()
```

Comment:
As this is probably called with higher frequency we should probably make this a normal for loop and directly do it in the copy constructor.